### PR TITLE
add simai.training_data.list(filters=...)

### PR DESCRIPTION
**Pull request by awoimbee**
[sc-24682]

This will unlock https://github.com/ansys/pysimai/pull/56#discussion_r1668211334

### CODE REVIEW

**Review by tmpbeing — COMMENTED** [review]
Nice feature but it needs documentation

**Comment by awoimbee** [comment]
Sphinx really sucks with type annotations. I had a hard time just getting it to work and I hate the end result (it's unreadable):
image

**Review by mmeisso — APPROVED** [review]
Nice !

File: src/ansys/simai/core/data/types.py [hunk 1]
```diff
@@ -288,3 +286,31 @@ class SubsetEnum(str, Enum):
...
+Filters type for ``list()`` endpoints that support them.
+
+The simplified ``Filters`` syntax (only for ``EQ`` filters) is: ``{"name": "my-experiment"}``.
+The full syntax is: ``[("name", "LIKE", "%thingy%"), ("size", "GT", 9000)]``.
+In both cases, the conditions are ``AND`` together.
+"""
+
+
+def to_raw_filters(filters: Optional["Filters"]) -> Optional["RawFilters"]:
+    match filters:
```
Comment (tmpbeing) [inline src/ansys/simai/core/data/types.py:310]: Can't use match, we still support Python 3.9

**Pull request merged**